### PR TITLE
Host resource bugfix

### PR DESCRIPTION
**Pull request by Sakshi-dell**
# Description
- port type is now computed based on value of port name
- add, remove and modify are now performed separately in update.
- Added validation that chap mutual username and password can not be present without single username and password.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Acceptance tests

### CODE REVIEW

File: powerstore/resource_host.go [hunk 1]
```diff
@@ -531,64 +508,144 @@ func (r resourceHost) planToServer(plan, state models.Host) *gopowerstore.HostMo
...
 	}
 
 	return hostUpdate
 }
 
 // to perform modify operation in update
-func (r resourceHost) modifyOperation(plan, state models.Host) *gopowerstore.HostModify {
-	hostUpdate := &gopowerstore.HostModify{}
+func (r resourceHost) modifyOperation(plan, state models.Host) []gopowerstore.UpdateInitiatorInHost {
+	//hostUpdate := &gopowerstore.HostModify{}
```
Comment (shenda1) [inline powerstore/resource_host.go]: Pls remove the comment if its not required

Comment (Sakshi-dell) [reply powerstore/resource_host.go]: removed

**Review by shenda1 — APPROVED** [review]

**Comment by prajwalpatil25** [comment]
please add coverage for resource_host.go

**Comment by Sakshi-dell** [comment]
image

**Review by prajwalpatil25 — APPROVED** [review]

**Pull request merged**